WeartGloves/src/WeArtGlove.cpp

```diff
@@ -0,0 +1,1598 @@
+    }
+    else
+    {
+        int textureIdParam = config.find("textureId").asInt32();
+        if (textureIdParam < 0 || textureIdParam > 22)
+        {
+            yError()<<LogPrefix<<"textureId should be between 0 and 21";
+            return false;
+        }
+        
+        textureId = (TextureType) textureIdParam;
+        yInfo()<<LogPrefix<<"textureId is:"<<textureIdParam;
+    }
+
+    // Get smoothing factor
+    if (!(config.check("smoothing_factor")))
+    {
+        yWarning()<<LogPrefix<<"couldn't find smoothing factor (alpha), setting smoothing_factor as default value"<< alpha;
+    }
+    else 
+    {
+        alpha = config.find("smoothing_factor").asFloat64();
+        if (alpha <= 0 || alpha > 1)
+        {
+            yError()<<LogPrefix<<"smoothing factor is not acceptable"<<alpha;
+            return false;
+        }    
+    }
+
+    // Get finger length
+    if (!(config.check("finger_length_mm")))
+    {
+        yWarning()<<LogPrefix<<"couldn't find finger length, setting finger_length as default value"<< fingerLength;
+    }
+    else 
+    {
+        fingerLength = config.find("finger_length_mm").asFloat64();
+        if (fingerLength <= 0)
+        {
+            yError()<<LogPrefix<<"finger length is not acceptable"<<fingerLength;
+            return false;
+        }
+    }
+
+    // Get Robot Joint limit
+    // MIN
+    yarp::os::Bottle* robotJointLimitMinYarp;
+    if (!(config.check("min_joint_limit")) || !config.find("min_joint_limit").isList())
+    {
+        yError()<<LogPrefix<<"couldn't find robot joint limits, check the xml file";
+        return false;
+    }
+    robotJointLimitMinYarp = config.find("min_joint_limit").asList();
+
+    for (int i = 0; i < robotJointLimitMinYarp->size(); i++)
+    {
+        MIN_JOINT_LIMIT(i) = robotJointLimitMinYarp->get(i).asFloat64();
+    }
+
+    // MAX
+    yarp::os::Bottle* robotJointLimitMaxYarp;
+    if (!(config.check("max_joint_limit")) || !config.find("max_joint_limit").isList())
+    {
+        yError()<<LogPrefix<<"couldn't find robot joint limits, check the xml file";
+        return false;
+    }
+    robotJointLimitMaxYarp = config.find("max_joint_limit").asList();
+
+    for (int i = 0; i < robotJointLimitMaxYarp->size(); i++)
+    {
+        MAX_JOINT_LIMIT(i) = robotJointLimitMaxYarp->get(i).asFloat64();
+    }
+
+    // Runs and Starts the Client and connect it to the Server (Middleware)
+    {
+        {
+            std::lock_guard<std::mutex> lock(weArtClientMutex);
+            weArtClient->Run();
+            while (!weArtClient->IsConnected())
+            {
+                yError() << "The weArtClient is not connected!";
+                return false;
+            }
+            weArtClient->Start();
+            // Start receiving raw sensor data from the middleware
+            weArtClient->StartRawData();
+            weArtClient->StartCalibration();
+            yarp::os::Time::delay(3);
```

Comment:
What is this delay for?

Reply:
Well, when calling `StartCalibration()`, the middleware needs few seconds for internal calibration and then starts the device. I noticed that withoud this delay the yarp device crashes. This delay seemed to fixed the issue.

Reply:
Is there an API call to check if the calibration is over? 3 is a bit of a magic number and I am afraid that it may not be enough either in same cases